French translation

gawi opened the pull request:
French translation for Globle.
Added a few mechanisms to be able to translate everything.
Added support for locale contry aliases.

gawi commented on src/components/Info.tsx:
```diff
@@ -146,8 +146,8 @@ export default function Info({ setScreen }: Props) {
-  // Need to skip question 6 if not English because it doesn't apply.
-  if (localeContext.locale !== "en-CA") {
+  // Need to skip question 6 if not English or French because it doesn't apply.
+  if (localeContext.locale !== "en-CA" && localeContext.locale !== "fr-CA") {
```
There is some support for aliases in French.

gawi commented on src/components/List.tsx:
```diff
@@ -59,7 +60,7 @@ export default function List({ guesses, win, globeRef }: Props) {
       {orderedGuesses.length > 0 && (
         <p className="my-1">
-          <b>{qualifier}</b>
+          <b><FormattedMessage id={qualifier} /></b>
```
"Works" but not sure if that's the way to go.

gawi commented on src/i18n/messages/es-MX.ts:
```diff
@@ -64,6 +64,8 @@ export const Spanish: Messages = {
   Settings7: "Idioma",
   Settings8: "Próximamente Globle Edición de Ciudades",
+  Answer: "Respuesta",
+  Closest: "Los países más cercanos",
```
Provisional translation. Please review. Not sure this is the appropriate wording in Spanish.

gawi commented:
Just realized another French translation has been submitted. How embarassing! I didn't really want to compete.

the-abe-train commented:
Bonjour, @gawi! There are indeed 2 French translations, and I merged the other PR for the only reason that it came first. However, I notice that your Quebecois, and I believe the other translation offered was from France. If you like, take a look at the 2 translations and, if you feel that the game would improve from having both, I'll merge this request as well. Let me know what you think!

gawi commented:
Hello @the-abe-train. I've fixed some minor French translation issues. Nothing at this point would justify a fr-CA.  Otherwise, I've added translation support for some recently introduced messages and added per-language alternate names support.

gawi commented on src/components/Statistics.tsx:
```diff
@@ -104,7 +104,7 @@ export default function Statistics({ setShowStats }: Props) {
   const event = new Date();
   // @ts-ignore
-  const unambiguousDate = event.toLocaleDateString("en-CA", options);
+  const unambiguousDate = event.toLocaleDateString(locale, options);
```
When sharing result with others, I'm assuming we want to format according to current locale. However, if we share to the entire planet, let's say, on Twitter, some users might prefer to use English.

gawi commented on src/data/alternate_names.json:
```diff
@@ -1,38 +1,66 @@
+  "es-MX": [],
+  "de-DE": [],
+  "pt-BR": []
+}
```
There must be a way to make this more type-safe.  It's easy to forget to add something here when adding a new language.

gawi commented on src/i18n/messages/fr-FR.ts:
```diff
@@ -8,57 +8,57 @@ export const French: Messages = {
   Stats4: "Série actuelle",
   Stats5: "Série max",
-  Stats6: "Moyenne de tentative",
-  Stats7: "Avg. Guesses",
```
Here, the original translator left it in English probably because he want to share his stats of the day in English.  I haven't found anything similar in the other translations.

gawi commented on src/pages/Info.tsx:
```diff
@@ -140,8 +140,8 @@ export default function Info() {
-  // Need to skip question 6 if not English because it doesn't apply.
-  if (localeContext.locale !== "en-CA") {
+  // Need to skip question 6 if not English or French because it doesn't apply.
+  if (localeContext.locale !== "en-CA" && localeContext.locale !== "fr-FR") {
```
I'm adding fr-FR here because there are some alternate names. Maybe you just want to get rid of this check and presume all translations will provide alternate names.
